src/GalleryTools/Commands/BackfillCommand.cs

```
@@ -251,15 +313,26 @@ private static async Task<string> GetFlatContainerUri(Uri serviceDiscoveryUri)
             return result.First().AbsoluteUri.TrimEnd('/');
         }
 
-        private async Task<TMetadata> FetchMetadata(HttpClient httpClient, string nuspecUri)
+        private async Task<TMetadata> FetchMetadataAsync(HttpClient httpClient, string nupkgUri, NuspecReader nuspecReader)
         {
-            using (var nuspecStream = await httpClient.GetStreamAsync(nuspecUri))
-            {
-                var document = LoadDocument(nuspecStream);
+            var httpZipProvider = new HttpZipProvider(httpClient);
+            httpZipProvider.RequireAcceptRanges = false;
+            httpZipProvider.RequireContentRange = false;
 
-                var reader = new NuspecReader(document);
-
-                return ReadMetadata(reader);
+            try
+            {
+                var zipDirectoryReader = await httpZipProvider.GetReaderAsync(new Uri(nupkgUri));
+                var zipDirectory = await zipDirectoryReader.ReadAsync();
+                var files = zipDirectory
+                    .Entries
+                    .Select(x => x.GetName())
+                    .ToList();
+
+                return ReadMetadata(files, nuspecReader);
+            }
+            catch (Exception)
+            {
+                return default; // fail silently without a value
```

We should understand these errors instead of failing silently.

---

Logging now.